lighthouse-core/config/index.js

```diff
@@ -171,16 +172,79 @@ function filterAudits(audits, auditWhitelist) {
   return filteredAudits;
 }
 
-function expandAudits(audits) {
+function expandAudits(audits, paths) {
+  const rootPath = path.join(__dirname, '../../');
+
   return audits.map(audit => {
-    try {
-      return require(`../audits/${audit}`);
-    } catch (requireError) {
+    // Check each path to see if the audit can be located. First match wins.
+    const AuditClass = paths.reduce((definition, auditPath) => {
+      // If the definition has already been found, just propagate it. Otherwise try a search
+      // on the path in this iteration of the loop.
+      if (definition !== null) {
+        return definition;
+      }
+
+      const requirePath = auditPath.startsWith('/') ? auditPath : path.join(rootPath, auditPath);
+      try {
+        return require(`${requirePath}/${audit}`);
+      } catch (requireError) {
+        return null;
+      }
+    }, null);
+
+    if (!AuditClass) {
       throw new Error(`Unable to locate audit: ${audit}`);
     }
+
+    // Confirm that the audit appears valid.
+    const auditValidation = validateAudit(AuditClass);
+    if (!auditValidation.valid) {
+      const errors = Object.keys(auditValidation)
+          .reduce((errorList, item) => {
+            // Ignore the valid property as it's generated from the other items in the object.
+            if (item === 'valid') {
+              return errorList;
+            }
+
+            return errorList + (auditValidation[item] ? '' : `\n - ${item} is missing`);
+          }, '');
+
+      throw new Error(`Invalid audit class: ${errors}`);
+    }
+
+    return AuditClass;
   });
 }
 
+function validateAudit(auditDefinition) {
+  const hasAuditMethod = typeof auditDefinition.audit === 'function';
```

Comment:
one other idea: this method could also be more something like `assertValidAudit()` and throw directly instead of passing an object back out. Would skip a lot of the object building only to concat it all back together